app/cmd/plugin_search.go

```diff
@@ -81,45 +84,60 @@ func matchPluginsData(plugins []client.AvailablePlugin) (result []client.CenterP
 	}
 	getCurrentJenkinsAndClient(&(jclient.JenkinsCore))
 	site, err := jclient.GetSite()
+	hasSite := false
 	if err != nil {
-		return
+		hasSite = true
 	}
-	pluginJclient := &client.PluginManager{
-		JenkinsCore: client.JenkinsCore{
-			RoundTripper: pluginSearchOption.RoundTripper,
-		},
-	}
-	getCurrentJenkinsAndClient(&(pluginJclient.JenkinsCore))
-	plu, err := pluginJclient.GetPlugins()
+	installedPlugins, err := pluginJclient.GetPlugins()
+	hasInstalledPlugin := false
 	if err != nil {
-		return
+		hasInstalledPlugin = true
 	}
-
 	for _, plugin := range plugins {
-		for _, updatePlugin := range site.UpdatePlugins {
-			if plugin.Name == updatePlugin.Name {
-				result = append(result, updatePlugin)
-				break
+		isTrue := false
+		if !hasSite {
+			if len(site.UpdatePlugins) > 0 && site != nil {
```

Comment:
It's wrong here. If site is nil, then panic will occur.

Reply:
I have tested when the received site is 500 Status code